gnsi/acltrie/trie.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package acltrie contains a trie for gNMI path with a corresponding policy.
+package acltrie
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	pathzpb "github.com/openconfig/gnsi/pathz"
+)
+
+// Trie is the root of the ACL trie.
+type Trie struct {
+	root *trieNode
+}
+
+type trieNode struct {
+	// childrenByRank orders the children of the node by number the non-wildcard keys in the child.
+	// This allows fast lookup of the best match: the one with the largest number of non-wildcard keys.
+	childrenByRank []map[string]*trieNode
```

Comment:
Suggest to repeat this comment here since people might read this first and ask what a rank is.
`// The rank of the child is the number of non-wildcard keys (if any).`

Reply:
done